### PR TITLE
Patches for allowing static build for PyPI distribution (pip install)

### PR DESCRIPTION
**Pull request by ProfFan**
Should not have impact on CI.

---
This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/207)

### CODE REVIEW

File: cmake/GtsamCythonWrap.cmake [hunk 1]
```diff
@@ -5,13 +5,17 @@ unset(PYTHON_EXECUTABLE CACHE)
 unset(CYTHON_EXECUTABLE CACHE)
 unset(PYTHON_INCLUDE_DIR CACHE)
 unset(PYTHON_MAJOR_VERSION CACHE)
+unset(PYTHON_LIBRARY CACHE)
```
Comment (dellaert) [inline cmake/GtsamCythonWrap.cmake]: GTSAM_USE_CUSTOM_PYTHON_LIBRARY?

File: cython/setup.py.in [hunk 2]
```diff
@@ -5,13 +5,13 @@ try:
...
     from distutils.core import setup, find_packages
 
-if 'SETUP_PY_NO_CHECK' not in os.environ:
-    script_path = os.path.abspath(os.path.realpath(__file__))
-    install_path = os.path.abspath(os.path.realpath(os.path.join('${GTSAM_CYTHON_INSTALL_PATH}${GTSAM_BUILD_TAG}', 'setup.py')))
-    if script_path != install_path:
-        print('setup.py is being run from an unexpected location: "{}"'.format(script_path))
-        print('please run `make install` and run the script from there')
-        sys.exit(1)
+# if 'SETUP_PY_NO_CHECK' not in os.environ:
```
Comment (dellaert) [inline cython/setup.py.in]: What is this commented out code?

File: cmake/GtsamCythonWrap.cmake (hunk 1 above)
Comment (ProfFan) [reply cmake/GtsamCythonWrap.cmake]: This is to unset the cache for PYTHON_LIBRARY (libpython paths)

File: cython/setup.py.in (hunk 2 above)
Comment (ProfFan) [reply cython/setup.py.in]: It is to check whether `setup.py` is being ran in the correct directory. I think this is unnecessary and is not common practice...

Comment (dellaert) [reply cython/setup.py.in]: Should we just kill it then? @thduynguyen any advice?

Comment (ProfFan) [reply cython/setup.py.in]: I'll kill it :)

Comment (ProfFan) [reply cython/setup.py.in]: NUKED

**Pull request merged**